modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/WhitelistInstanceBinding.java

```diff
@@ -47,15 +49,21 @@
      */
     public final List<String> canonicalTypeNameParameters;
 
+    /** The {@link Map} of annotations for this field. */
+    public final Map<String, String> annotations;
```

Comment:
I would do this a bit differently. A couple suggestions:

* have a WhitelistAnnotation marker interface
* have annotation parsers keyed by the string we look for in the whitelist, eg deprecation
* on the whitelist objects, we can have a Map<Class<? extends Annotation>, ? extends Annotation>, so that we can look up annnotations by type
* pass the map of parsers to the whitelist loader, and have a static base annotations that can be used, eg with `no_import` and in the future `deprecated`
* pass the annotation parser a Map<String, String>, ie make the contents of the parsed annotation a keyed map like java has. this will allow future expansion and clarity with named parameters instead of positional

Reply:
Thanks for the feedback.  Will get to work with the changes we discussed.